spec/rubocop/cop/style/redundant_regexp_character_class_spec.rb

```diff
@@ -307,6 +307,15 @@
     end
   end
 
+  context 'with a character class containing an escaped-0' do
+    # See https://github.com/rubocop/rubocop/issues/11067 for details - in short "\0" != "0" - the
+    # former means an Unicode code point `"\u0000"`, the latter a number character `"0"`.
+    # Similarly "\032" means "\u001A". Other numbers starting with "\0" can also be mentioned.
+    it 'does not register an offense' do
+      expect_no_offenses('foo = /[\032]/')
```

Comment:
Will redundant codes, like `[\032\032]` or `[\b\b]`, register an offense? If not, shouldn't they?

Reply:
The latter _does_ register an offense presently, so I suppose the former will as well?

```
% cat test3.rb                                                       
puts "one \010 two \010".gsub(/[\b]/, "potato")

puts "one \010 two \010".gsub(/[\b\b]/, "potato")

% ruby test3.rb
one potato two potato
one potato two potato

% rubocop --only Lint/DuplicateRegexpCharacterClassElement -- test3.rb
Inspecting 1 file
W

Offenses:

test3.rb:3:35: W: [Correctable] Lint/DuplicateRegexpCharacterClassElement: Duplicate element inside regexp character class
puts "one \010 two \010".gsub(/[\b\b]/, "potato")
                                  ^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

Reply:
Ah, what this PR fixes is `Style/RedundantRegexpCharacterClass`. So, it looks like I was looking at a different cop that was similar. I will see about `Lint/DuplicateRegexpCharacterClassElement` in a separate PR.

Reply:
I've opened #11086.